test/hotspot/jtreg/runtime/cds/appcds/jcmd/JCmdTest.java

```diff
@@ -0,0 +1,294 @@
+                argsContainOpts(args, new String[] {"-XX:+RecordDynamicDumpInfo", "-XX:ArchiveClassesAtExit="})) {
+                // app exit premature due to incompactible args
+                return null;
+            }
+            Process proc = app.getProcess();
+            if (e instanceof IOException && proc.exitValue() == 0) {
+                // Process started and exit normally.
+                return null;
+            }
+            throw e;
+        }
+        return app;
+
+    }
+
+    private static void test(String jcmdSub, String archiveFile,
+                             long pid, boolean expectOK) throws Exception {
+        boolean isStatic = jcmdSub.equals(SUBCMD_STATIC_DUMP);
+        String fileName = archiveFile != null ? archiveFile :
+            ("java_pid" + pid + (isStatic ? "_static" : "_dynamic") + ".jsa");
+        File file = new File(fileName);
+        if (file.exists()) {
+            file.delete();
+        }
+
+        String jcmd = "VM.cds " + jcmdSub;
+        if (archiveFile  != null) {
+          jcmd +=  " " + archiveFile;
+        }
+
+        PidJcmdExecutor cmdExecutor = new PidJcmdExecutor(String.valueOf(pid));
+        OutputAnalyzer output = cmdExecutor.execute(jcmd, true/*silent*/);
+
+        if (expectOK) {
+            output.shouldHaveExitValue(0);
+            if (!file.exists()) {
+                throw new RuntimeException("Could not create shared archive: " + fileName);
+            } else {
+                file.delete();
+            }
+        } else {
+            if (file.exists()) {
+                throw new RuntimeException("Should not create shared archive " + fileName);
+            }
+        }
+    }
+
+    private static void print2ln(String arg) {
+        System.out.println("\n" + arg + "\n");
+    }
+
+    // Those two flags will not create a successful LingeredApp.
+    private static String[] noDumpFlags  =
+        {"-XX:+DumpSharedSpaces",
+         "-Xshare:dump"};
+    // Those flags will be excluded in static dumping.
+    private static String[] excludeFlags =
+        {"-XX:DumpLoadedClassList=AnyFileName.classlist",
+         "-XX:+UseSharedSpaces",
+         "-XX:+RecordDynamicDumpInfo",
+         "-XX:+RequireSharedSpaces",
+         "-XX:+DynamicDumpSharedSpaces",
+         "-XX:ArchiveClassesAtExit=tmp.jsa",
+         "-Xshare:auto",
+         "-Xshare:on"};
+
+    // Times to dump cds against same process.
+    private static final int ITERATION_TIMES = 2;
+
+    private static void test() throws Exception {
+        LingeredApp app  = null;
+        long pid;
+
+        // 1. Static dump with default name multiple times.
+        print2ln("1: Static dump with default name multiple times.");
+        app  = createLingeredApp("-cp", jarFile);
+        pid = app.getPid();
+        for (int i = 0; i < ITERATION_TIMES; i++) {
+            test(SUBCMD_STATIC_DUMP, null, pid, EXPECT_PASS);
+        }
+        app.stopApp();
```

Comment:
For successful dumping cases like the above, you may want to add a runtime case making sure one of the classes in the jar is loaded from the archive.